File: leetcode-swift/lc_001.swift
//
// LeetCode
// 1. Two Sum
// https://leetcode.com/problems/two-sum
//


class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numsDict: [Int: Int] = [:]

        for index in 0..<nums.count {
            let complement = target - nums[index]

            if let position = numsDict[complement] {
                return [position, index]
            }

            numsDict[nums[index]] = index
        }

        return [-1, -1]
    }
}

File: leetcode-go/lc_001.go
//
// LeetCode
// 1. Two Sum
// https://leetcode.com/problems/two-sum
//


func twoSum(nums []int, target int) []int {
    indices := []int{-1, -1}

    for i:= 0; i < len(nums) - 1; i++ {
        for j := i + 1; j < len(nums); j++ {
            if nums[i] + nums[j] == target {
                indices[0] = i
                indices[1] = j
                break
            }
        }
    }

    return indices
}
